packages/ui-components/src/components/MenuItem/MenuItem.component.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useContext, MouseEvent, FC } from "react"
+import { Menu as HLMenu } from "@headlessui/react"
+import { MenuContext } from "../Menu/Menu.component"
+import { Icon, KnownIconsEnum } from "../Icon/Icon.component"
+
+const itemStyles = `
+   jn-text-theme-default
+   jn-flex
+   jn-items-center
+   jn-w-full
+   cursor-pointer
+   bg-clip-padding
+   jn-truncate
+   jn-text-left
+   jn-bg-theme-background-lvl-1
+   disabled:jn-cursor-not-allowed
+   data-[headlessui-state="disabled"]:jn-cursor-not-allowed
+ `
+
+const smallStyles = `
+   jn-text-sm
+   jn-p-2
+ `
+
+const normalStyles = `
+   jn-text-base
+   jn-pt-[0.6875rem]
+   jn-pb-[0.5rem]
+   jn-px-[0.875rem]
+ `
+
+const actionableItemStyles = `
+   hover:jn-bg-theme-background-lvl-3
+   data-[headlessui-state="disabled"]:jn-bg-theme-background-lvl-3
+ `
+
+export interface MenuItemProps {
+  /** Children of the menu item */
+  children?: React.ReactNode
+  /** Pass a custom className to the menu item */
+  className?: string
+  /** Whether the menu item is disabled */
+  disabled?: boolean
+  /** Pass the name of an icon the button should show. Can be any icon included with Juno. */
+  icon?: keyof typeof KnownIconsEnum
+  /** The label of the menu item. Will take precedence over children passed to the component. */
+  label?: string
+  /** Pass an href to the menu item. Will result in the menu item being rendered as an `<a>`. */
+  href?: string
+  /** Pass an onClick handler to the menu item. Will result in the menu item being rendered as a `<button>`. */
+  onClick?: (_event: MouseEvent<HTMLButtonElement>) => void
```

Comment:
I would suggest to put 
// eslint-disable-next-line no-unused-vars this instead of _event name variables with underscore

Reply:
Done. I could remove the underscore, and now there is no unused-vars error anymore (there used to be one).

Reply:
but in a code it is stilll here did you push your changes?

Reply:
Yes I did push, the underscore is removed in the code below – not sure what you are seeing?

Here's how the upstream code on this PR looks to me right now:

<img width="657" alt="Screenshot 2024-11-06 at 10 44 36" src="https://github.com/user-attachments/assets/00fdf9bc-fb55-4687-932c-af8e3ac28a7a">

Reply:
Ah, sorry, yes, you are right, will remove in an instant (line 56).

Reply:
Done. My apologies for the confusion.